### ISSUE TITLE
refresh on playback position change

### ISSUE THREAD

**Issue by oaprograms** [jaruba/wcjs-player#20]
Hello,

One minor issue, I don't know if it's easy to fix or not: when you change playback position while the player is paused, the picture doesn't update.

Also, thank you for this awesome library. I am using it in my new open source project, a video player for learning languages. Please check it out and let me know what you think:
https://github.com/oaprograms/lingo-player

Also, I love your code on GitHub, Powder player is amazing. More people should be using it (i.e. everyone). I didn't have time to dig into its code, but I will. I hope we can get in touch, maybe even work on something together.

Ognjen

**Comment by jaruba** [comment]
> One minor issue, I don't know if it's easy to fix or not: when you change playback position while the player is paused, the picture doesn't update.

Uhm, I think I might have caused this issue in `WebChimera.js` with [this request](https://github.com/RSATom/WebChimera.js/issues/35), as VLC was sending the same frame at 24fps even when paused. (and I saw no reason for it to do that until now, it just seemed inefficient to use so much memory for drawing the same frame as it was paused)

@RSATom what do you think? :)

> Also, thank you for this awesome library.

Your welcome, I'm glad you found it useful. :)

> I am using it in my new open source project, a video player for learning languages. Please check it out and let me know what you think: https://github.com/oaprograms/lingo-player

That looks awesome, would you mind if we add it to the [Known Projects using WebChimera.js List](https://github.com/RSATom/WebChimera.js#known-projects-using-webchimerajs)?

> Also, I love your code on GitHub, Powder player is amazing. More people should be using it (i.e. everyone).

Thanks, I agree, but I haven't really advertised it anywhere.. I'm waiting to release v1.0 (which is taking forever to make..) before I think of places where I can engage people in using it.

> I didn't have time to dig into its code, but I will.

Might seem messy as I didn't use any templating system (such as angular) when building it.

> I hope we can get in touch, maybe even work on something together.

You can find my email address on my github profile page.

**Comment by jaruba** [comment]
@RSATom I'm wondering if this couldn't be fixed in `WebChimera.js` while still maintaining the efficiency of not drawing the same frame when paused.

I'm thinking of some logic like:

```
if (paused) {
    if (position != savedPausePosition) {
        savedPausePosition = position;
        // draw the frame
    }
}
```

**Comment by RSATom** [comment]
@jaruba, I'll check if it related to mentioned above patch.

**Comment by RSATom** [comment]
@jaruba, checked - this patch is definitely the reason.

**Comment by RSATom** [comment]
maybe we should back to idea to handle such special cases in js...

**Comment by RSATom** [comment]
published new version with completely removed frame update ignore.

**Comment by jaruba** [comment]
@RSATom ah, saw you published a new `WebChimera.js` version, I'll test out possible solutions to this in `wcjs-renderer`, you should of removed the black frame patch too, so we can do that in the renderer too.

**Comment by RSATom** [comment]
black frame was removed in one of previous releases, I just forgot about it.

**Comment by jaruba** [comment]
great :)

**Comment by RSATom** [comment]
when you will implement your black frame show, keep in mind, `frameReady` could come even after `onEndReached`. It's known issue.

**Comment by jaruba** [comment]
Ah, thanks for mentioning it, I can block rendering when paused (and position not changed) and ended in `wcjs-renderer` now..

**Comment by oaprograms** [comment]
Sorry for late reply...

> That looks awesome, would you mind if we add it to the Known Projects using WebChimera.js List?

I wouldn't of course, thanks.

> Thanks, I agree, but I haven't really advertised it anywhere.. I'm waiting to release v1.0 (which is taking forever to make..) before I think of places where I can engage people in using it.

Powder looks pretty stable, I've been using it for a long time, with no issues, so maybe you should 'advertise' it more.

**Comment by jaruba** [comment]
> Powder looks pretty stable, I've been using it for a long time, with no issues, so maybe you should 'advertise' it more.

Powder Player v0.90 (the current version) is the most buggy and inefficient version I ever released because I had to completely switch from WebChimera QML Player to WebChimera.js Player in that version. The difference between using a NPAPI Plugin as opposed to the new `WebChimera.js.node` was huge code wise, the great amount of changes turned into a lot of bugs ofc. (mostly minor bugs, but still bugs)

It's been 3 months since I released v0.90 and I think I managed to squash most of the bugs and improve the experience quite a bit for the next version.
